src/Stratis.Features.FederatedPeg/TargetChain/CrossChainTransferStore.cs

```diff
@@ -537,75 +554,78 @@ public Task<Transaction> MergeTransactionSignaturesAsync(uint256 depositId, Tran
             {
                 lock (this.lockObj)
                 {
-                    this.Synchronize();
-
-                    this.logger.LogInformation("ValidateCrossChainTransfers : {0}", depositId);
-                    ICrossChainTransfer transfer = this.ValidateCrossChainTransfers(this.Get(new[] { depositId })).FirstOrDefault();
-
-                    if (transfer == null)
+                    lock (((FederationWalletManager)this.federationWalletManager).lockObject)
                     {
-                        this.logger.LogInformation("FAILED ValidateCrossChainTransfers : {0}", depositId);
+                        Guard.Assert(this.Synchronize());
```

Comment:
There also appears to be other changes in this PR like this which seems significant to program flow. Maybe we need to look at separating the changes into smaller PRs.

Reply:
Just being consistent by having `Guard.Assert` everywhere `Synchonize` is used. Previously we just assumed success. See the summary for what is addressed in the PR.

Reply:
See the summary for what has been done in this PR. This is the only seemingly unrelated change but its relatively minor.